File: typecast-php/tests/Unit/RemoveSilenceTest.php
<?php
declare(strict_types=1);
namespace Neosapience\Typecast\Tests\Unit;

use Neosapience\Typecast\Models\Output;
use Neosapience\Typecast\Models\OutputStream;
use PHPUnit\Framework\TestCase;

final class RemoveSilenceTest extends TestCase
{
    public function testOptionalZeroAndRange(): void
    {
        foreach ([Output::class, OutputStream::class] as $class) {
            self::assertArrayNotHasKey('remove_silence_ms', (new $class())->toArray());
            foreach ([0, 300, 1000] as $value) {
                self::assertSame($value, (new $class(removeSilenceMs: $value))->toArray()['remove_silence_ms']);
            }
            foreach ([-1, 1001] as $value) {
                try {
                    (new $class(removeSilenceMs: $value))->toArray();
                    self::fail('Invalid silence accepted');
                } catch (\InvalidArgumentException $error) {
                    self::assertStringContainsString('removeSilenceMs', $error->getMessage());
                }
            }
        }
    }
}

File: typecast-ruby/test/remove_silence_test.rb
require "minitest/autorun"
require "typecast"

class RemoveSilenceTest < Minitest::Test
  def test_optional_zero_and_range
    [Typecast::Models::Output, Typecast::Models::OutputStream].each do |klass|
      refute klass.new.to_h.key?(:remove_silence_ms)
      [0, 300, 1000].each { |ms| assert_equal ms, klass.new(remove_silence_ms: ms).to_h[:remove_silence_ms] }
      [true, false, "100", 100.0, -1, 1001].each do |ms|
        assert_raises(ArgumentError) { klass.new(remove_silence_ms: ms) }
      end
    end
  end
end
